src/diagnostics/MOM_debugging.F90

```diff
@@ -837,67 +838,101 @@ end subroutine chksum_vec_A2d
+  real, dimension(HI%isd:,HI%jsd:,:), intent(in) :: salinity    !< The salinity field to sum [S ~> ppt] or [ppt]
   character(len=*),                   intent(in) :: mesg        !< An identifying message
+  type(unit_scale_type),    optional, intent(in) :: US       !< A dimensional unit scaling type
+  real,                     optional, intent(in) :: H_to_MKS !< A constant that translates thickness units to its
+                                                             !! MKS units (m or kg m-2) based on whether the model is
+                                                             !! Boussinesq [m H-1 ~> 1] or not [kg m-2 H-1 ~> 1]
   ! NOTE: This subroutine uses "save" data which is not thread safe and is purely for
   ! extreme debugging without a proper debugger.
-  real, save :: totalH = 0.   ! The total ocean volume, saved for the next call [m3]
-  real, save :: totalT = 0.   ! The total volume integrated ocean temperature, saved for the next call [degC m3]
-  real, save :: totalS = 0.   ! The total volume integrated ocean salinity, saved for the next call [ppt m3]
+  real, save :: totalH = 0.   ! The total ocean volume or mass, saved for the next
+                              ! call [H L2 ~> m3 or kg] or [m3] or [kg]
+  real, save :: totalT = 0.   ! The total volume integrated ocean temperature, saved for the next
+                              ! call [C H L2 ~> degC m3 or degC kg] or [degC m3] or [degC kg]
+  real, save :: totalS = 0.   ! The total volume integrated ocean salinity, saved for the next
+                              ! call [S H L2 ~> ppt m3 or ppt kg] or [ppt m3] or [ppt kg]
   ! Local variables
   logical, save :: firstCall = .true.
-  real, dimension(HI%isc:HI%iec, HI%jsc:HI%jec) :: tmp_for_sum ! The volume of each column [m3]
-  real :: thisH, delH  ! The total ocean volume and the change from the last call [m3]
-  real :: thisT, delT  ! The current total volume integrated temperature and the change from the last call [degC m3]
-  real :: thisS, delS  ! The current total volume integrated salinity and the change from the last call [ppt m3]
+  real :: tmp_for_sum(HI%isc:HI%iec, HI%jsc:HI%jec) ! The volume of each column [H L2 ~> m3 or kg] or [m3] or [kg]
+  real :: thisH, delH  ! The total ocean volume and the change from the last call [H L2 ~> m3 or kg] or [m3] or [kg]
+  real :: thisT, delT  ! The current total volume integrated temperature and the change from the last
+                       ! call [C H L2 ~> degC m3 or degC kg] or [degC m3] or [degC kg]
+  real :: thisS, delS  ! The current total volume integrated salinity and the change from the last
+                       ! call [S H L2 ~> ppt m3 or ppt kg] or [ppt m3] or [ppt kg]
+  real :: H_unscale    ! A constant that translates thickness units to its MKS units (m or kg m-2) based on
+                       ! whether the model is Boussinesq [m H-1 ~> 1] or not [kg m-2 H-1 ~> 1]
```

Comment:
Remove "not"?

Reply:
I have changed `not` to `non-Boussinesq` for clarity.

Reply:
ah!  OK, I get it now.  Sorry about that.